appserver/payara-appserver-modules/healthcheck-checker/src/main/java/fish/payara/healthcheck/mphealth/MicroProfileHealthChecker.java

```diff
@@ -244,26 +284,32 @@ private URI buildURI(Server server, String endpoint) throws URISyntaxException {
             String sysPropsPort = configProps.getPropertyValue(basePort);
             truePort = Integer.parseInt(sysPropsPort);
         }
-        return new URI(protocol, null, TranslatedConfigView.expandValue(listener.getAddress()), truePort, "/" + endpoint, null, null);
+        return new URI(protocol, null, TranslatedConfigView.expandValue(listener.getAddress()), truePort,
+                "/" + endpoint, null, null);
     }
 
-    //send request to remote healthcheck endpoint to get the status
-    private HealthCheckResultEntry pingHealthEndpoint(String instanceName, URI remote) {
-        Client jaxrsClient = ClientBuilder.newClient();
-        WebTarget target = jaxrsClient.target(remote);
```

Comment:
NB: Replaced this with pure `HttpURLConnection` usage as it was failing for reasons unrelated to the task itself. Some problem deep in the config of JAX-RS and its dependencies. I figured that if complexity can make this fail the fix is to get rid of it so I did.

Reply:
Only a note: Client has close method, so not closing it is a resource leak. But it does not implement Closeable, so tools do not report it. I learned it when I create JMH performance tests in previosu company and ended in OOME ;-)

Reply:
The another problem is hunting us again and again - ClientBuilder uses ServiceLocator so it usually finds implementation from the server ... with tracing etc. Custcom have currently several issues around this.
So if you can use HttpURLConnection directly, it is perhaps even better solution in this case :-)